tests/models/test_match_spec.py

```diff
@@ -717,3 +717,60 @@ def test_parse_build_number_brackets(self):
         #     "build_number": '>=3',
         # }
 
+
+class MatchSpecMergeTests(TestCase):
+
+    def test_merge_single_name(self):
+        specs = (MatchSpec('exact'), MatchSpec('exact 1.2.3 1'), MatchSpec('exact >1.0,<2'))
+        merged_specs = MatchSpec.merge(specs)
+        print(merged_specs)
+        assert len(merged_specs) == 1
+        merged_spec = merged_specs[0]
+        print(merged_spec)
+        assert str(merged_spec) == "exact[version='1.2.3,>1.0,<2',build=1]"
+        assert merged_spec.match({
+            'name': 'exact',
+            'version': '1.2.3',
+            'build': '1',
+            'build_number': 1,
+        })
+        assert not merged_spec.match({
+            'name': 'exact',
+            'version': '1.2.2',
+            'build': '1',
+            'build_number': 1,
+        })
+
+    def test_merge_multiple_name(self):
+        specs = tuple(MatchSpec(s) for s in (
+            'exact', 'exact 1.2.3 1', 'exact >1.0,<2',
+            'bounded >=1.0,<2.0', 'bounded >=1.5', 'bounded <=1.8',
+        ))
+        merged_specs = MatchSpec.merge(specs)
+        print(merged_specs)
+        assert len(merged_specs) == 2
+
+        exact_spec = next(s for s in merged_specs if s.name == 'exact')
+        bounded_spec = next(s for s in merged_specs if s.name == 'bounded')
+
+        assert str(exact_spec) == "exact[version='1.2.3,>1.0,<2',build=1]"
```

Comment:
This is fine.  It's more constraints than it needs to be.  My code cut them out, but I expect this is safer than my code (i.e. less likely to lose the finer details of when < and <= are not equivalent), and less prone to logic errors.  It's a little confusing looking at it, because so much of the information there is superfluous - you have to think about what bound or pin will take effect, rather than just seeing it.  That said, as long as the performance impact of having more constraints will scale well, I'm happy enough with how this is.

Reply:
Yeah this is what @mcg1969 suggested, exactly because of the logical complexity of reducing to a minimal form.  I think this is the solution we should go with for now, with the goal of implementing the minimal-form reduction in the future.